### ISSUE TITLE
glob actions need better documentation

### ISSUE THREAD

**Issue by ericoporto** [actions/toolkit#685]
Glob actions are referenced for pattern matching throughout the actions (like https://github.com/marketplace/actions/upload-a-build-artifact) but the docs don't explain how the pattern engine works. There's no guidance on how it should work and even the tests are undewhelming in providing documentation on how it should work.

If this is using ANY regex engine or other engine leveraged from nodejs or other thing, please SAY SO and point to the docs of that engine. If those docs don't exist one needs to be written. Right now, it's really hard to understand how pattern matching should work, and this is a very useful feature, specially in workflows that uses multiple OSes and environments in their strategy/matrix.

**Comment by criemen** [comment]
It is especially confusing that the glob action doesn't correspond to the publicly documented glob feature used in the `on.push.paths` property:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

I think it would be great to have the glob-action do exactly what is done on `on.push.paths`.

**Comment by alex-page** [comment]
I would also like to know if there are any benefits for using `actions/glob` over another globbing npm package. Is this better for use with GitHub actions? Why?

**Comment by StephenHodgson** [comment]
yeah I have no idea if a pattern like `${{ steps.artifacts.outputs.releaseDirectory }}/**/*.{aab,apk,obb,zip}` would even work or not until I test it. 

... Yeah doesn't work 😅
